recipes/emsdk_installer/all/test_package/conanfile.py

```diff
@@ -0,0 +1,18 @@
+from conans import ConanFile, CMake, tools, RunEnvironment
+import os
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "build_type", "arch", "compiler"
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self, generator='MinGW Makefiles' if os.name == 'nt' else 'Unix Makefiles', parallel=False)
+        cmake.definitions["CONAN_DISABLE_CHECK_COMPILER"] = True
+        cmake.configure()
+        cmake.build()
+        #self.run("cmake --build %s --config Release" % self.build_folder)
+
+    def test(self):
+        test_file = os.path.join(self.build_folder, "bin", "test_package.js")
+        self.run('node %s' % test_file, run_environment=True)
```

Comment:
Conan-center-index has a `nodejs` recipe.
Perhaps you can add a build requirement on `nodejs` (only needed in the test package)?

Reply:
There is a problem. The emsdk downloads its own nodejs and wants to use it. The recipe in bincrafters used a trick to supress this. But in the current versions I think since 2.0 its kinda hard. And it feels like fighting with emsdk :smiley: .
I want to provide a solution which allows people to use emsdk to compile other packages to emscripten wasm. Even if it is not so good to have dependencies not managed by conan. I want to let the dependencies problem open to a fix by somebody else with more motivation in this part. I am open to fix the your other remarks.

Reply:
Let's first fix the errors in https://github.com/conan-io/conan-center-index/pull/6163#issuecomment-873920734 so c3i can at least start building something.